### PR TITLE
qt4-mac: fixes for macOS 12 Monterey

### PR DESCRIPTION
**Pull request by michaelld**
#### Description

Amazingly qt4-mac builds and executes on the macOS 12 Monterey beta! Just a small tweak / patchfile to fix the build with the latest Clang. I added a few other minor tweaks that technically aren't related.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.0 21A5294g x86_64
Xcode 12.5.1 12E507

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by kencu** [comment]
Looks like the same thing qt did to fix it themselves

https://github.com/qt/qtscript/commit/97ec1d1882a83c23c91f0f7daea48e05858d8c32

SO I would imagine it would be the proper fix for newer compilers (not having actually tested it as yet :> ).

I suppose the acid test would be to build qt4 on Tiger and see if it still works with this -- I can do that in a few days, as I'm away from my Tiger machines at present.

**Comment by michaelld** [comment]
Interesting ... I'll have to follow that repo now :) There's no rush here ; there aren't a lot of ports that require Qt4 any longer, and macOS 12.0 is still months away

**Comment by michaelld** [comment]
@kencu on https://build.macports.org/waterfall do you know how to track down a specific port? I'm still a N00B there ...

**Comment by kencu** [comment]
> @kencu on https://build.macports.org/waterfall do you know how to track down a specific port? I'm still a N00B there ...

I sometimes try searching the page for the port I want, and have patience to try the next page or two, but often that will not show up the issue.

The best way I have found so far is to use the linked builds from the github commit interface.

Screen Shot 2021-07-30 at 8 26 35 PM

Sometimes, like this example, there is no actual build here (the checkmark only indicates the buildbot liked the commit). But if there is a red "x" then you have a failure and the link is to the failing build log.

Those red "x" and green "check" marks go back a long way. Even if the actual build log is no longer available, the marks often remain to show you what succeeded and failed.

**Review by kencu — APPROVED** [review]

**Comment by michaelld** [comment]
OK so really not different than what I do ... well worth asking! I'll let you merge this once you've tested on Tiger.

**Comment by kencu** [comment]
qt4-mac with this patchset is building away on i386 Tiger now.

**Comment by kencu** [comment]
successful build of this PR on TIger. Looks GTM.

**Pull request merged**
